### PR TITLE
test: codecov

### PR DESCRIPTION
**Pull request by dannyrandall**
check it out: https://app.codecov.io/gh/dannyrandall/copilot-cli

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by iamhopaul123 — APPROVED** [review]
This is cool!

File: .gitignore
Comment (efekarakus) [inline .gitignore:11]: nit: do we need to add coverage.out here as well? 🤷 just asking because you have one for the custom resources

File: README.md
Comment (efekarakus) [inline README.md]: :doge: I can't tell if this a good sell or not 😂 

Is there a way of modifying the `test-cov` job and setting a treshold such that we get a slack message when we are below a certain percentage? I don't want to block merging PRs but it would be good to know if we are below our current threshold in the future by getting a notification

File: .gitignore
Comment (dannyrandall) [reply .gitignore:11]: it's already there :blush:

File: README.md
Comment (rclinard-amzn) [reply README.md]: I agree with the second part but if it's too much work to add that to this PR then we can probably put it in a follow-up PR! Up to @dannyrandall though 😄

Comment (dannyrandall) [reply README.md]: ok so here's what i'm thinking! we'll upload reports as part of `ci/test` and on commits to mainline. then, when it runs on PRs, it'll comment on the PR with the changes like [here](https://github.com/aws/copilot-cli/pull/3926#issuecomment-1227716386).

Comment (rclinard-amzn) [reply README.md]: Sounds good! Will this comment be made every time there's commits? That could become spammy.

Comment (dannyrandall) [reply README.md]: i _think_ what it does is just edit the first comment it makes if it needs to update anything

**Review by huanjani — APPROVED** [review]
Bring on the finger-wagging!

**Pull request merged**
